### PR TITLE
SplunkPy - set the time period when fetch with offset

### PR DESCRIPTION
**Pull request by ilappe**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-17928

## The problem:
When we fetched notables from Splunk and the number of notables are over the `limit`
we iterate over the queue with `offset` in each iteration 
but in each offset iteration - we used the same `earliest_time` but `latest_time` increased and increased
this lead to lost incidents when the new latest_time contained new notables
in addition, currently, when we get # of notables less than the limit rate - we assume we reached out the last notables from the requested period, and this are wrong, because there may be 
dropped incidents, and if the number of dropped + number of new incidents are equal to the limit rate we should continue fetched with offset

## The fix:
send the query with the original `earliest_time` and `latest_time` was sent at the start of the queue
and if we get # (new incidents + dropped incidents) >= to the rate limit - fetch with offset.

### CODE REVIEW

**Review by DeanArbel — COMMENTED** [review]
LGTM, please see my comment below

File: Packs/SplunkPy/Integrations/SplunkPy/README.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ where the **$IDENTITY_VALUE** is replaced with the **user** and **src_user** fro
 #### How to configure
 1. Configure the integration to fetch incidents.
 2. *Enrichment Types*: Select the enrichment types you want to enrich each fetched notable with. If none are selected, the integration will fetch notables as usual (without enrichment).
-3. *Fetch events query*: The query for fetching events. The default query is for fetching notable events. You can edit this query to fetch other types of events. Note that to fetch notable events, make sure the query uses the \`notable\` macro.  
+3. *Fetch events query*: The query for fetching events. The default query is for fetching notable events. You can edit this query to fetch other types of events. Note that to fetch notable events, make sure the query uses the \`notable\` macro and the results are sorted by time (for example: add `| sort 0 _time` to the query).
```
Comment (ShirleyDenkberg) [inline Packs/SplunkPy/Integrations/SplunkPy/README.md:82]: Please replace the bolded text (I am not able to submit my changes for some reason.)
 ....  \`notable\` macro ** and the results are sorted by time**. See the default query as an example.

**Comment by ShirleyDenkberg** [comment]
@ilappe For some reason I am not able to submit my edit.  So, on line 153 in the README file, please replace the text with the bold text as shown below.

You can go to the *Fetch events query* parameter and select the query to fetch the notables from Splunk. Make sure to provide a query which uses the \`notable\` macro **and the results are sorted by time**. See the default query as an example.

@DeanArbel @esharf Doc review completed.

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 2]
```diff
@@ -315,6 +315,9 @@ def build_fetch_query(dem_params):
             field_trimmed = field.strip()
             fetch_query = fetch_query + ' | eval ' + field_trimmed + '=' + field_trimmed
 
+    if 'sort 0 _time' not in fetch_query:
```
[the remark this answers is not in the document]
Comment (ilappe) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:318]: only way 
as discussion - if user put sort, we add our sort at the end of query with a pipe
and it work well

**Comment by ilappe** [comment]
@DeanArbel 
added some performance improvements

**Review by DeanArbel — APPROVED** [review]
Great job!

**Review by DeanArbel — APPROVED** [review]
Please update the title of this PR to match your changes

**Pull request merged**